src/test/java/com.teragrep.jos_01.procfs/ProcessTest.java

```diff
@@ -0,0 +1,355 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import com.teragrep.jos_01.procfs.status.process.Stat;
+import com.teragrep.jos_01.procfs.status.process.Statm;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class ProcessTest {
+
+    private final Logger LOGGER = LoggerFactory.getLogger(ProcessTest.class);
+    Process systemd;
+    Process kthreadd;
+
+    public static class TestThread implements Runnable {
+
+        private final ProcessBuilder processBuilder;
+        private java.lang.Process process;
+
+        public TestThread(ProcessBuilder processBuilder) {
+            this.processBuilder = processBuilder;
+        }
+
+        @Override
+        public void run() {
+            try {
+                this.process = processBuilder.start();
+            }
+            catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private ArrayList<Integer> getPidByCommand(String command) {
+        ArrayList<Integer> pids = new ArrayList<Integer>();
+        Assertions.assertDoesNotThrow(() -> {
+
+            java.lang.Process ps = Runtime.getRuntime().exec("ps");
```

Comment:
I'd generally immediately reject any external calls in tests but I guess here it is unavoidable. `ps` is not guaranteed to be installed so it might fail on containers, can you add rextra check that it actually exists?